artemis-protocols/artemis-mqtt-protocol/pom.xml

```diff
@@ -111,6 +111,22 @@
          <artifactId>log4j-slf4j-impl</artifactId>
          <scope>test</scope>
       </dependency>
+      <dependency>
+         <groupId>org.apache.activemq.tests</groupId>
+         <artifactId>artemis-test-support</artifactId>
+         <version>${project.version}</version>
+         <scope>test</scope>
+      </dependency>
```

Comment:
I missed it previously but this dep is problematic. That module is aimed at for use within the tests/ subtree only currently and like all those module it isnt deployed, so we shouldnt be using it in modules that are being deployed.

I guess perhaps you used it as a proxy for artemis-server:test-jar due to prior discussion on that and eventually getting rid of the test-jar, but it doesnt really work in this case unfortunately.

Of course, it only seems to be getting used to have ActiveMQTestBase as a base class for 1 test, which it then uses to stand up a full VM server, so arguably the test should be in the tests/integration-tests (or stretching maybe tests/unit-tests) anyway rather than the mqtt-protocol module since its not really a classical unit test to start with. Then it could indeed depend on artemis-test-support as those bits already do.

That or change to the actual test-jar dep and add it to the cleanup list hehe.

Reply:
This should be fixed now. I went in a different direction with this whole test.